test/verifiedTests/testSolvers/testOptimizeCbModelNLP.m

```diff
@@ -0,0 +1,63 @@
+% The COBRAToolbox: testSolveCobraNLP.m
+%
+% Purpose:
+%     - tests the optimizeCbModelNLP function, and some of its parameters.
+%
+% Authors:
+%     - Original file: * Joseph Kang 11/16/09
+%                      * Richard Que (02/11/10) NLP Support
+%     - CI integration: Laurent Heirendt, March 2017
+%
+% Note:
+%     - The tomlab_snopt solver must be tested with a valid license
+
+% save the current path
+currentDir = pwd;
+
+% initialize the test
+initTest(fileparts(which(mfilename)));
+
+% set the tolerance
+tol = 1e-6;
+
+load('ecoli_core_model','model')
+
+% define the solver packages to be used to run this test
+solverPkgs = {'glpk'};  % tomlab_snopt
+
+for k = 1:length(solverPkgs)
+
+    % change the COBRA solver (NLP)
+    solverOK = changeCobraSolver(solverPkgs{k});
```

Comment:
The default functions use glpk to determine a basic solution for initialization of the NLP optimization.